src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,246 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/FilterDecoratorFeaturizer.h"
+#include "Components/GrainFeaturizerImpl.h"
+#include "Components/WindowFeaturizerBase.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         LagLeadOperatorTransformer
+///  \brief         Copy values from prior data or future data
+///                 Input type is a tuple of std::vector<std::string>, representing grains, and std::double_t, representing target column
+///
+template <typename InputT>
+class LagLeadOperatorTransformer :
```

Comment:
Please add these for transformer as ort need them to allocate memory for matrix:
std::ptrdiff_t getRowsNumber() const {return _offsets.size();}
std::ptrdiff_t getColsNumber() const {return _horizon;}

Reply:
This solution won't scale to dynamically shaped matrix types; we need to design this in a more scalable way.

---
In reply to: [395172177](https://github.com/microsoft/FeaturizersLibrary/pull/171#discussion_r395172177) [](ancestors = 395172177)

Reply:
sure